tmt/base.py

```diff
@@ -972,88 +1082,139 @@ def show(self) -> None:
+    def lint_unknown_keys(self) -> LinterReturn:
+        """ T001: all keys are known """
 
-        # File does not exist
-        if not manual_test.exists():
-            return verdict(False, f"file '{self.test}' does not exist")
+        # We don't want adjust in show/export so it is not yet in Test._keys
+        invalid_keys = self._lint_keys(EXTRA_TEST_KEYS + OBSOLETED_TEST_KEYS + ['adjust'])
 
-        # Check syntax for warnings
-        warnings = tmt.export.check_md_file_respects_spec(manual_test)
-        if warnings:
-            for warning in warnings:
-                verdict(False, warning)
-            return False
+        if invalid_keys:
+            for key in invalid_keys:
+                yield LinterOutcome.FAIL, f'unknown key "{key}" is used'
 
-        # Everything looks ok
-        return verdict(True, f"correct manual test syntax in '{self.test}'")
+            return
 
-    def lint(self) -> bool:
-        """
-        Check test against the L1 metadata specification.
+        yield LinterOutcome.PASS, 'correct keys are used'
 
-        Return whether the test is valid.
-        """
-        self.ls()
-        assert self.path is not None  # narrow type
-
-        # Check test, path and summary (use bitwise '&' because 'and' is
-        # lazy and would skip all verdicts following the first fail)
-        valid = verdict(
-            bool(self.test), 'test script must be defined')
-        valid &= verdict(
-            self.path.is_absolute(), 'directory path must be absolute')
-        if self.path.is_absolute():
-            assert self.path is not None  # narrow type
-            test_path = Path(self.node.root) / self.path.unrooted()
-        else:
-            test_path = self.path
-        valid &= verdict(
-            test_path.exists(), 'directory path must exist')
-        self._lint_summary()
+    def lint_defined_test(self) -> LinterReturn:
+        """ T002: test script must be defined """
+
+        if not self.test:
+            yield LinterOutcome.FAIL, 'test script is not defined'
+            return
+
+        yield LinterOutcome.PASS, 'test script is defined'
+
+    def lint_absolute_path(self) -> LinterReturn:
+        """ T003: test directory path must be absolute """
+
+        if not self.path:
+            yield LinterOutcome.FAIL, 'directory path is not set'
+            return
+
+        if not self.path.is_absolute():
+            yield LinterOutcome.FAIL, 'directory path is not absolute'
+            return
+
+        yield LinterOutcome.PASS, 'directory path is absolute'
+
+    def lint_path_exists(self) -> LinterReturn:
+        """ T004: test directory path must exist """
+
+        if not self.path:
+            yield LinterOutcome.FAIL, 'directory path is not set'
+            return
+
+        test_path = Path(self.node.root) / self.path.unrooted()
+
+        if not test_path.exists():
+            yield LinterOutcome.FAIL, f'test path "{test_path}" does not exist'
```

Comment:
```suggestion
            yield LinterOutcome.FAIL, f"test path '{test_path}' does not exist"
```

Let's consistently use [single quotes](https://tmt.readthedocs.io/en/latest/contribute.html#introduction) when reporting messages to the user.

Reply:
Fixed in 21c4e01c.